sbomtool/pkg/generate/generate.go

```diff
@@ -0,0 +1,136 @@
+package generate
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/format/cyclonedxjson"
+	"github.com/anchore/syft/syft/format/spdxjson"
+	"github.com/anchore/syft/syft/sbom"
+)
+
+// closeFile properly closes a file and handles any close errors.
+// It logs close errors to stderr but doesn't return them to avoid
+// overriding more important errors from the main operation.
+func closeFile(f *os.File, operation string) {
+	if closeErr := f.Close(); closeErr != nil {
+		slog.Error("Failed to close file", "operation", operation, "error", closeErr)
+	}
+}
+
+// createSpdxSbom creates an SPDX SBOM file
+func createSpdxSbom(sbom sbom.SBOM, name string, outDir string) error {
+	cfg := spdxjson.DefaultEncoderConfig()
+	cfg.Pretty = false
+	cfg.Version = "2.3"
```

Comment:
The latest SPDX versions for SBOMs seems to be 3.0.0:

https://spdx.dev/use/specifications/

Could you please elaborate on why supporting only 2.3? Also, if we will support multiple versions at the time, an abstraction over the versions could be needed here.

Reply:
We need to standardize on the version because each is a schema change (and therefore a different SBOM file) In this case, both Trivy and Syft support 2.3, but not 3.0